packages/core/src/config.ts

```
@@ -3,33 +3,133 @@ import path from "path";
+const DEFAULT_CONFIG: Sinc.Config = {
   sourceDirectory: "src",
   rules: [],
   includes,
   excludes,
   tableOptions: {}
 };
 
-export const DEFAULT_CONFIG_FILE: string = `
-module.exports = {
-  sourceDirectory: "src",
-  rules: [],
-  excludes:{},
-  includes:{},
-  tableOptions:{}
-};
-`.trim();
+let ConfigManager = new (class {
+  private root_dir: string | undefined;
+  private config: Sinc.Config | undefined;
+  private manifest: SN.AppManifest | undefined;
+  private config_path: string | undefined;
+  private source_path: string | undefined;
+  private env_path: string | undefined;
+  private manifest_path: string | undefined;
+  constructor() {}
+
+  async loadStartupFiles() {
+    //Ordered due to config dependencies within loads
+    let skipConfigPath = false; //Prevents logging error messages during init
+    await loadConfigPath()
+      .then(_config_path => {
+        if (_config_path) this.config_path = _config_path;
+        else skipConfigPath = true;
+      })
+      .then(() => {
+        return loadRootDir(skipConfigPath);
+      })
+      .then(_root_dir => {
+        if (_root_dir) this.root_dir = _root_dir;
+      })
+      .then(() => {
+        return loadConfig(skipConfigPath);
+      })
+      .then(_config => {
+        if (_config) this.config = _config;
+      })
+      .then(loadEnvPath)
+      .then(_env_path => {
+        if (_env_path) this.env_path = _env_path;
+      })
+      .then(loadSourcePath)
+      .then(_source_path => {
+        if (_source_path) this.source_path = _source_path;
+      })
+      .then(loadManifestPath)
+      .then(_manifest_path => {
+        if (_manifest_path) this.manifest_path = _manifest_path;
+      })
+      .then(loadManifest)
+      .then(_manifest => {
+        if (_manifest) this.manifest = _manifest;
+      })
+      .catch(e => {
+        throw e;
+      });
+  }
+
+  getConfig(setup = false) {
+    if (this.config) return this.config;
+    logger.error("Error getting config");
```

We probably want to throw an error on these lines to stop execution